lib/seattleflu/db/cli/command/longitudinal.py

```diff
@@ -0,0 +1,278 @@
+                      .agg(lambda x: x.tolist()) \
+                      .reset_index() \
+                      .drop(stub, axis=1) \
+                      .rename(columns={f'new_{stub}': stub})
+
+
+def weekly_assessments_wide_to_long(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Casts a given DataFrame *df* from wide to long on longitudinal data.
+    """
+    rename_map = {}
+    swab_columns = list(df.filter(regex='swab_[0-9]+$', axis=1))
+    for col in swab_columns:
+        rename_map[col] = re.sub("swab", "swab_collected", col)
+
+    df = df.rename(columns=rename_map)
+
+    stubnames = longitudinal_stubnames(df)
+
+    return pd.wide_to_long(df, stubnames, i='study_id', j="week", sep="_",
+                           suffix='\\w+'
+                          ).reset_index()
+
+
+def longitudinal_stubnames(df: pd.DataFrame) -> [str]:
+    """
+    Returns a list of the stubnames of longitudinal columns that have suffixes
+    depicting the week number of the study (e.g. swab_date_11)
+
+    TODO: better optimized way to apply this, e.g. via `filter`?
+    """
+    stubnames = []
+    for column in list(df):
+        match = re.match(".*(?=_[0-9]+$)", column)
+        if match:
+            stubnames.append(match[0])
+    return list(set(stubnames))
+
+
+def duplicate_audere_keys(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    For a given DataFrame *df* of longitudinal data, duplicate every column
+    whose meaning is a 1:1 match with Audere data. The duplicated column will
+    match the Audere naming conventions for easier queries down the road.
+    """
+    sch_to_audere_map = {
+        "study_id": "individual",
+        "age": "age",
+        "sex": "AssignedSex",
+        "race": "Race",
+        "ethnicity": "HispanicLatino",
+        "insurance": "MedicalInsurance",
+        "home_type": "WhereLive",
+        "bedrooms": "Bedrooms",
+        "house_members": "PeopleInHousehold",
+        "smoking": "HouseholdSmoke",
+        "enroll_site": "site",
+        "flu_shot": "FluShot",
+        "sx_specific": "Symptoms",
+        "illness_impact": "DailyInterference",
+        "travel_type": "ChildrensRecentTravel",
+        "sample_id": "barcode",
+        "assess_date": "encountered",
+    }
+
+    for key in sch_to_audere_map:
+        try:
+            df[sch_to_audere_map[key]] = df[key]
+        except:
+            LOG.warning(f"Column {key} not found in the given data.")
+
+    return df
+
+
+def insert_static_values(df: pd.DataFrame) -> pd.DataFrame:
+    """ Inserts static values into a given DataFrame *df* """
+    df["ChildrenNearChildren"] = "yes"
+    df["SchoolType"] = "childcareCenter"
+    df["ChildrenHutchKids"] = np.where(df['enroll_site'] == 1, 'yes', 'no')
+
+    return df
+
+
+def hash_pii(df: pd.DataFrame) -> pd.DataFrame:
+    """ Hashes personally identifiable information in a given DataFrame *df*. """
+    df['study_id'] = df['study_id'].astype('str').apply(generate_hash)
```

Comment:
I just looked at the data SCH sent us and I'm a little worried that just hashing the `study_id` alone will generate too many repeats since the `study_id` is just a number

Reply:
I was also curious if I was being "too secure" by hashing `study_id`.

Reply:
I think we would want to keep things consistent and make the `individual.identifier` still be a 64 character string. We might just want to hash multiple things about the individual so that we don't get repeated identifiers.

Reply:
What's a good strategy for selecting these columns?
I've reworked this with `study_id` + `race` + `census_tract`.

Reply:
I'm not sure we need to hash anything here, although we do need to add additional bits of entropy/information to make it unique in a wider space of values.

Stepping back, the goal is a consistent identifier for the individual, which `study_id` already is. As Jover pointed out, the problem with using that alone is that it is unqualified. We could solve this by namespacing it with static values, for example: `redcap/sch/year-1/childcare/<study_id>`. This is similar (but with a longer namespace prefix) to what we did with the NWGC internal ids we use as identifiers for the presence/absence data (`NWGC_<id>`).